### PR TITLE
Refactors detail view in advance of trivia

### PR DESCRIPTION
**Pull request by RedBrogdon**
This removes the seasons circles and some other kruft from the details view in favor or a more traditional hierarchy. Four new data fields have been added to each veggie record so there's more to display. A segmented control has been added to the details page as well, so users can flip between info and trivia questions.

The trivia bit is forthcoming. If you have an amusing question about artichokes, for example, please suggest it.

### CODE REVIEW

**Comment by RedBrogdon** [comment]
Preview of new UI:

preview2

**Comment by RedBrogdon** [comment]
@willlarche, do you mind taking a look at that screenshot and letting me know if I'm using the segmented control in an appropriate manner?

**Comment by xster** [comment]
tangential: it seems kinda weird hierarchically that the save to garden is at the bottom of the screen. Perhaps add it to the top banner's upper right corner and use https://github.com/flutter/flutter/pull/29630?

**Comment by RedBrogdon** [comment]
I can't use anything not in the current stable release, so it'd be awhile before I could import that icon. I'm very open to changing how things are marked as favorites, though. I don't like how that switch looks tacked on at the end.

**Comment by xster** [comment]
Ah ya, that's a good point. Please disregard then.

File: veggieseasons/lib/screens/details.dart [hunk 1]
```diff
@@ -6,68 +6,222 @@ import 'package:flutter/cupertino.dart';
...
+      future: targetCalories,
+      builder: (context, snapshot) {
+        final target = snapshot?.data ?? 2000;
+        final percent = standardPercentage * 2000 ~/ target;
 
-  String get _firstChars {
-    return '${season.toString().substring(7, 8).toUpperCase()}'
-        '${season.toString().substring(8, 9)}';
+        return Text(
+          '$percent% DV',
```
Comment (xster) [inline veggieseasons/lib/screens/details.dart]: Is DV a common word people know?

File: veggieseasons/lib/screens/details.dart [hunk 2]
```diff
@@ -6,68 +6,222 @@ import 'package:flutter/cupertino.dart';
...
+              style: Styles.detailsServingHeaderText,
             ),
           ),
-        ),
+          Container(
+            decoration: BoxDecoration(
+              border: Border.all(color: Styles.servingInfoBorderColor),
+            ),
+            padding: const EdgeInsets.all(8.0),
+            child: Column(
```
Comment (xster) [inline veggieseasons/lib/screens/details.dart]: This build is kinda long. Maybe break this into another widget?

**Comment by willlarche** [comment]
LGTM!

File: veggieseasons/lib/screens/details.dart (hunk 1 above)
Comment (RedBrogdon) [reply veggieseasons/lib/screens/details.dart]: I've seen it in multiple sources while looking up nutrition facts. I'll update the note below the values to provide a hint, though.

File: veggieseasons/lib/screens/details.dart (hunk 2 above)
Comment (RedBrogdon) [reply veggieseasons/lib/screens/details.dart]: Tables add so much rightward drift. :(

I'll break it out.

**Pull request merged**